src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -76,7 +89,23 @@ public static boolean unlock(final Device dev) {
             }
             n--;
         }
-        return false;
+        if (!isUnlocked) {
+            n = numOfBits;
+            while (n > 0) {
+                for (CharSequence perm : perms) {
+                    doPeek(perm);
+                    doPoke();
+                    // we want a random number of spins from 1 to the number of bits. 0 spins is invalid doSpin()
+                    isUnlocked = doSpin((int)Math.round(Math.random() * numOfBits) + 1);
+                    if (isUnlocked) {
+                        break;
+                    }
+                }
+                n--;
+            }
+        }
+        if (isUnlocked) appendTrace("device is unlocked");
```

Comment:
This is more nit picky than anything, but I think it is a better coding practice to use brackets instead of omitting them. I would double check just to be sure.

Reply:
Good eye! It's a very bad habit of mine. I will edit this as a TODO in my bugfixes branch.